backend/reviewduck/src/main/java/com/reviewduck/review/domain/ReviewForm.java

```diff
@@ -48,76 +54,122 @@ public class ReviewForm extends BaseDate {
+
+        this.questions.subList(0, beforeSize).clear();
+
+        sortQuestions();
     }
 
-    private void sortQuestions(List<ReviewFormQuestion> reviewFormQuestions) {
-        int index = 0;
-        for (ReviewFormQuestion reviewFormQuestion : reviewFormQuestions) {
-            reviewFormQuestion.setPosition(index++);
+    private ReviewFormQuestion createOrUpdateQuestion(ReviewFormQuestionUpdateDto question) {
+        Optional<ReviewFormQuestion> existedQuestion = questions.stream()
+            .filter(it -> question.getId() != null && Objects.equals(it.getId(), question.getId()))
+            .findAny();
+
+        if (existedQuestion.isPresent()) {
+            ReviewFormQuestion updatedQuestion = existedQuestion.get();
+            updatedQuestion.update(question.getValue(), question.getDescription());
+            return updatedQuestion;
         }
+
+        return new ReviewFormQuestion(question.getValue(), question.getDescription(), this);
     }
 
-    private void validateWhenCreate(Member member, String title, List<ReviewFormQuestion> questions) {
+    public boolean isMine(Member member) {
+        return this.member.equals(member);
+    }
+
+    private void validateWhenCreate(Member member, String title, List<ReviewFormQuestionCreateDto> questions) {
         validateBlankTitle(title);
         validateTitleLength(title);
         validateNullMember(member);
         validateNullQuestions(questions);
     }
 
-    private void validateWhenUpdate(String title, List<ReviewFormQuestion> questions) {
+    private void validateWhenUpdate(String title, List<ReviewFormQuestionUpdateDto> questions) {
         validateBlankTitle(title);
         validateTitleLength(title);
         validateNullQuestions(questions);
+        validateQuestionExist(questions);
+    }
+
+    private void validateBlankTitle(String title) {
+        if (title == null || title.isBlank()) {
+            throw new ReviewFormException("회고 폼의 제목은 비어있을 수 없습니다.");
+        }
+    }
+
+    private void validateTitleLength(String title) {
+        if (title.length() > 100) {
+            throw new ReviewFormException("회고 폼의 제목은 100자를 넘을 수 없습니다.");
+        }
     }
 
     private void validateNullMember(Member member) {
-        if (Objects.isNull(member)) {
+        if (member == null) {
             throw new ReviewFormException("회고 폼의 작성자가 존재해야 합니다.");
         }
     }
 
-    private void validateNullQuestions(List<ReviewFormQuestion> questionValues) {
-        if (Objects.isNull(questionValues)) {
+    private void validateNullQuestions(List<?> questionValues) {
+        if (questionValues == null) {
             throw new ReviewFormException("회고 폼의 질문 목록 생성 중 오류가 발생했습니다.");
         }
     }
 
-    private void validateTitleLength(String title) {
-        if (title.length() > 100) {
-            throw new ReviewFormException("회고 폼의 제목은 100자를 넘을 수 없습니다.");
-        }
+    private void validateQuestionExist(List<ReviewFormQuestionUpdateDto> questions) {
```

Comment:
이름은 exist인데 실제로는 파라미터로 들어온 dto의 question들에 대해 id가 null이 아닌지와 내부에 ReviewForm에 존재하는 question인지 검증하고 있는 것 같아요. 존재여부를 검증하는 함수처럼 느껴지는데, validateQuestionUpdatable 등의 이름은 어떨까요? 내부에서 어떤 검증 과정을 거치는지 메서드네이밍에서 드러낼 필요는 없는 것 같아요!

Reply:
그렇군요. 메서드명이 책임을 명확하게 드러내지 않았던 것 같아서 수정하였습니다. 

```suggestion
    private boolean isQuestionUpdatable(ReviewFormQuestionUpdateDto questionToUpdate) {

```